flocker/control/httpapi.py

```diff
@@ -185,6 +185,134 @@ def saved(ignored):
+        #     raise NotImplemented()
+
+        # if metadata is None:
+        #     raise NotImplemented()
+
+        # Get the current configuration
+        # deployment = self.persistence_service.get()
+
+        # Find all manifestations of dataset
+        # manifestations = manifestations_from_deployment(deployment, dataset_id)
+
+        # if not manifestations:
+        #     raise NotFound(
+        #         'Dataset not found. '
+        #         'There are no manifestations of the requested dataset. '
+        #         'dataset_id: {}'.format(dataset_id)
+        #     )
+
+        # Lookup the node that has a primary Manifestation (if any)
+
+        # expected_manifestation = Manifestation(dataset=dataset, primary=True)
+        # primary_nodes = list(
+        #     node for node in deployment.nodes if node.hostname == primary
+        # )
+        # if len(primary_nodes) == 0:
+        #     # No primary node
+        #     # raise something here?
+        #     # Spec says not
+        #     # "ie, don't accept non-IP addresses but do accept IP addresses that aren't part of the cluster"
+              # Ah there's already a comment about that....
+
+              # XXX Check cluster state to determine if the given primary node
+              # actually exists.  If not, raise PRIMARY_NODE_NOT_FOUND.
+              # See FLOC-1278
+        #     primary_node = None
+        # else:
+
+        #     # New primary was found.
+        #     # There should only be one.
+
+        #     (primary_node,) = primary_nodes
+
+        # if expected_manifestation in primary_node.manifestations():
+
+        #     # No change needed. return early?
+        #     pass
+
+        # Now construct a new_deployment where the primary manifestation of the dataset is on the requested primary node
+        # But what if the dataset is associated with an Application? Should the Application be moved to the new node too?
+        # And do we need to mark the manifestation as a replica on the existing primary node?
+        # Or will that be up to the convergence agent and state API
+
+        # new_node_config = Node(
+        #     hostname=primary_node.hostname,
+        #     applications=primary_node.applications,
+        #     other_manifestations=(
+        #         primary_node.other_manifestations | frozenset({manifestation})
+        #     )
+        # )
+        # other_nodes = frozenset(
+        #     node for node in deployment.nodes if node is not primary_node
+        # )
+        # new_deployment = Deployment(
+        #     nodes=other_nodes | frozenset({new_node_config})
+        # )
+        # saving = self.persistence_service.save(new_deployment)
+
+        # Construct the return dictionary from the supplied dataset_id, primary
+        # and the found existing primary manifestation.
+        # What if there wasn't an existing primary manifestation?
+        # Should the returned and maximum size be that found on a lone replica?
```

Comment:
Don't worry about it for now, file a followup issue, since:

1. We don't have any code yet anywhere to create non-primary manifestations.
2. Our model can't yet express information about datasets that aren't node-specific. It will need to do, but there's no code yet.